src/server/scripts/Battlegrounds/SeethingShore/battleground_seething_shore.cpp

```diff
@@ -0,0 +1,820 @@
+                creature->AI()->DoAction(SeethingShore::Actions::CommanderText1);
+    }
+
+    void OnPrepareStage3() override
+    {
+        _scheduler.Schedule(2s, [this](TaskContext context)
+        {
+            if (_activeAzeriteNodes.size() < MAX_AZERITE_SPAWNS)
+            {
+                SpawnAzeriteNodes();
+                context.Repeat(100ms);
+            }
+            else
+                context.Repeat(2s);
+        });
+    }
+
+    void OnStart() override
+    {
+        for (ObjectGuid const& guid : _doors)
+        {
+            if (GameObject* gameObject = battlegroundMap->GetGameObject(guid))
+            {
+                gameObject->UseDoorOrButton();
+                gameObject->DespawnOrUnsummon(3s);
+            }
+        }
+
+        _scheduler.Schedule(30s, 2min, [this](TaskContext context)
+        {
+            SpawnBuffs();
+            context.Repeat();
+        });
+    }
+
+    void HandleAssaultPoint(GameObject* capturePoint, Player* player)
+    {
+        if (!player)
+            return;
+
+        capturePoint->SendCustomAnim(1);
+        Team const team = battleground->GetPlayerTeam(player->GetGUID());
+        TeamId const teamId = GetTeamIdForTeam(team);
+
+        if (team == ALLIANCE)
+        {
+            battleground->PlaySoundToAll(SeethingShore::Sounds::PvpFlagCapturedAlliance);
+            player->CastSpell(player, SeethingShore::Spells::CapturedAllianceCosmeticFx, true);
+        }
+        else if (team == HORDE)
+        {
+            battleground->PlaySoundToAll(SeethingShore::Sounds::PvpFlagCapturedHorde);
+            player->CastSpell(player, SeethingShore::Spells::CapturedHordeCosmeticFx, true);
+        }
+
+        battleground->AddPoint(team, 100);
+        UpdateWorldState(SeethingShore::WorldStates::AllianceScore, battleground->GetTeamScore(TEAM_ALLIANCE));
+        UpdateWorldState(SeethingShore::WorldStates::HordeScore, battleground->GetTeamScore(TEAM_HORDE));
+        player->CastSpell(player, SeethingShore::Spells::AchievementCredit, true);
+
+        // the achievement criteria doesn't have a aura condition so we do it manually here
+        if (player->HasAura(SeethingShore::Spells::AchievementTrackerCreditClaimJumper))
+            player->CastSpell(player, SeethingShore::Spells::AchievementCreditClaimJumper, true);
+
+        if (battleground->GetTeamScore(teamId) >= WARNING_NEAR_VICTORY_SCORE && _isInformedNearVictory[teamId])
+        {
+            _isInformedNearVictory[teamId] = true;
+            if (teamId == TEAM_ALLIANCE)
+                battleground->SendBroadcastText(SeethingShore::Texts::AllianceNearVictory, CHAT_MSG_BG_SYSTEM_NEUTRAL);
+            else
+                battleground->SendBroadcastText(SeethingShore::Texts::HordeNearVictory, CHAT_MSG_BG_SYSTEM_NEUTRAL);
+        }
+
+        battleground->UpdatePvpStat(player, SeethingShore::PvpStats::CapturedAzerite, 1);
+        capturePoint->SetAnimKitId(2560, false);
```

Comment:
I know it's hard, but any idea what the AnimKit does? Just to name it rather than using plain numbers.

Reply:
Animation that makes it "destroy" / "being mined", not really sure what naming would fit better